### PR TITLE
Serve pressed interactives

### PR DESCRIPTION
**Pull request by DavidLawes**
## What does this change?
This change allows us to serve pressed interactives to readers. Pressed interactives will only be served to readers if:
- the interactive-librarian  experiment is switched on
- the interactive path is defined in Set `interactives`

At the time of committing it is not possible to batch tag pressed interactives or programmatically add a tracking tag. While this solution is being worked on, we will keep a list of pressed interactives that Visuals have agreed to show to readers. When we have the ability to add tags to pressed interactives our solution will change. For now we also retain the experiment in the controller to enable us to quickly hide pressed content in the event of unexpected errors.

When we can tag pressed articles (tracking/dcroptout) then we will:
- tag all articles that are hardcoded in the frontend codebase
- remove file PressedInteractives entirely
- update the InteractiveController to show pressed pages based on presence of the tag
- update the press+clean functionality to automate tagging as part of this process

Some notes:
- Using the X-Accel-Redirect header redirects all requests to the s3 bucket defined here: https://github.com/guardian/platform/blob/main/router/files/router.conf#L677. This means that, if testing in non-Prod the files from the prod bucket will be served. This caused me a bit of confusion so thought I'd highlight this.
- I've updated the docs to add a little note about how to set switch overrides in DEV, hope my notes make sense.
- During testing in DEV and CODE I noticed some kind of race condition between the config loading and the interactive being served, meaning that on the first request of an interactive the relevant switch config appeared `off`, despite being turned `on` in the remote config. The result was that an unpressed page was displayed. During subsequent requests I could see the switch config appearing `on` - as expected. The issue I couldn't work out was that `ActiveExperiments.isParticipating(ShowPressedInteractives)` was consistently returning `false`, despite the underlying evaluations returning `true` (the switch itself was on and the server side experiment was switched on). Given my intention is to remove the switch altogether when we have confidence in the pressed pages being displayed to readers in PROD, I decided not to get too hung up on this point... Happy to try and think more about it though if this all sounds too implausible!

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## Screenshots
Screenshot 2021-09-21 at 15 54 38

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Comment by DavidLawes** [comment]
[Serve pressed interactives (intermediate solution)](https://trello.com/c/MrMiXAK5/374-serve-pressed-interactives-intermediate-solution)

**Comment by DavidLawes** [comment]
[Serve pressed interactives (intermediate solution)](https://trello.com/c/MrMiXAK5/374-serve-pressed-interactives-intermediate-solution)

File: common/app/experiments/Experiments.scala [hunk 1]
```diff
@@ -25,7 +25,7 @@ object LiveblogRendering
       participationGroup = Perc0A,
     )
 
-object PressedInteractives
+object ShowPressedInteractives
```
Comment (jlieb10) [inline common/app/experiments/Experiments.scala:28]: This is probably for some historic reason, but just double checking that this should be classed as an experiment?

Comment (DavidLawes) [reply common/app/experiments/Experiments.scala:28]: Yeah, good question - it was originally added as an experiment and I've kept it as one. In my mind it's still more of an experiment than a switch. I retained the experiment in this PR to allow us to easily turn off the serving of pressed content if we start to notice unexpected errors and quickly want to revert to rendered content. After a period of time (say 2 weeks), in which we've built up confidence in the solution, I'd propose deleting the experiment altogether. 

What do you think, does it make sense?

**Review by nicl — APPROVED** [review]

**Comment by DavidLawes** [comment]
[Serve pressed interactives (intermediate solution)](https://trello.com/c/MrMiXAK5/374-serve-pressed-interactives-intermediate-solution)

**Review by nicl — APPROVED** [review]
Thanks for the detailed description. 👍 looks good!

**Pull request merged**